### PR TITLE
mark upgradeable=false when unsupportedConfig set

### PR DESCRIPTION
**Pull request by jboyd01**

### CODE REVIEW

**Comment by jboyd01** [comment]
We want pr #33 to merge first
/hold

File: pkg/operator/starter.go [hunk 1]
```diff
@@ -18,6 +18,7 @@ import (
 	operatorv1informers "github.com/openshift/client-go/operator/informers/externalversions"
 	"github.com/openshift/library-go/pkg/controller/controllercmd"
 	"github.com/openshift/library-go/pkg/operator/status"
+	"github.com/openshift/library-go/pkg/operator/unsupportedconfigoverridescontroller"
```
Comment (jmrodri) [inline pkg/operator/starter.go:21]: HOLY MOLY! That's one long name :( I probably would've aliased this to ucoc or something shorter but that's more style than actual problem.

File: pkg/operator/starter.go [hunk 2]
```diff
@@ -130,6 +131,8 @@ func RunOperator(ctx *controllercmd.ControllerContext) error {
 		return err
 	}
 
+	configUpgradeableController := unsupportedconfigoverridescontroller.NewUnsupportedConfigOverridesController(operatorClient, ctx.EventRecorder)
```
Comment (jmrodri) [inline pkg/operator/starter.go:125]: Aliasing would've made the package shorter.

File: pkg/operator/starter.go [hunk 3]
```diff
@@ -139,6 +142,7 @@ func RunOperator(ctx *controllercmd.ControllerContext) error {
 	go clusterOperatorStatus.Run(1, ctx.Done())
 	go finalizerController.Run(1, ctx.Done())
 	go resourceSyncController.Run(1, ctx.Done())
+	go configUpgradeableController.Run(1, ctx.Done())
```
Comment (jmrodri) [inline pkg/operator/starter.go:136]: Interesting so we spawn a bunch of things on go routines.

**Review by jmrodri — APPROVED** [review]
On the surface this all looks good to me, an alias would be nice but not making that required.

**Comment by jboyd01** [comment]
e2e-aws
```
level=error msg="2 errors occurred:"
level=error msg="\t* module.vpc.aws_route_table_association.private_routing[5]: index 5 out of range for list aws_route_table.private_routes.*.id (max 5) in:"
level=error
level=error msg="${aws_route_table.private_routes.*.id[count.index]}"
level=error msg="\t* module.vpc.aws_route.to_nat_gw[5]: index 5 out of range for list aws_route_table.private_routes.*.id (max 5) in:"
level=error
level=error msg="${aws_route_table.private_routes.*.id[count.index]}"```
/retest

**Pull request merged**